kover-gradle-plugin/docs/index.md

```diff
@@ -85,223 +757,374 @@ buildscript {
 ```kotlin
-koverReport {
-    filters {
-        // filters for all reports
-    }
-  
-    verify {
-        // verification rules for all reports
+kover {
+    reports {
+        total {
+            verify {
+                // verification rules
+            }
+        }
     }
+}
+```
+
+### Specify verification rules for named variant
 
-    defaults {
-        xml { /* default XML report config */ }
-        html { /* default HTML report config */ }
-        verify { /* default verification config */ }
-        log { /* default logging config */ }
+Configuring verification rules for named report variant `foo`
+```kotlin
+kover {
+    reports {
+        variant("foo") {
+            verify {
+                // verification rules
+            }
+        }
     }
 }
 ```
 
-### Kotlin Android project
-These are projects that use Kotlin Android Gradle plugin.
+You can read more about creating verification rules in the [corresponding section](#verification).
 
-In such projects, a [variant of Kover tasks](#kover-android-tasks) is created for each Android build variant, 
-e.g. `koverHtmlReportRelease`, `koverXmlReportRelease`, `koverVerifyRelease` for `release` build variant.
+### Configuring report tasks
 
-Calling default report tasks (like `koverHtmlReport`) will result in a message `Task 'koverHtmlReport' will be skipped because no tests were executed` in the build log, and the report will not be generated.
+```kotlin
+filters {
+    // clear filters and override by specified 
+}
+
+filtersAppend {
+    // add filters
+}
+
+xml {
+    title = "Custom XML title"
+    onCheck = false
+    xmlFile = file("files")
+}
 
-To configure reports, the [android reports settings block](configuring#configuring-android-reports) is used:
+html {
+    title = "Custom HTML title"
+    onCheck = false
+    charset = "UTF-8"
+    htmlDir = dir("my-html")
+}
+
+log {
+    onCheck = false
+    header = "Header"
+    groupBy = GroupingEntityType.APPLICATION
+    aggregationForGroup = AggregationType.COVERED_PERCENTAGE
+    format = "<entity> line coverage: <value>%"
+    coverageUnits = CoverageUnit.LINE
+}
+
+binary {
+    onCheck = false
+    file = file("file.ic")
+}
+
+verify {
+    // clear verification rules and override by specified 
+}
+
+verifyAppend {
+    // add verification rules
+}
+```
+
+### Verification
+#### Coverage value
+During verification, the entire code is divided into units of code for which it determines whether it was covered (executed) or skipped (not executed).
+For example, an entire line from source code or a specific JVM instruction from compiled byte-code can be executed or not.
+
+All units are grouped into one or more groups.
+Based on amount of the executed and non-executed code units, one number (coverage value) will be calculated for each group using the aggregation function.
+
+Type `CoverageUnit` determines for which types of units the coverage will be measured.
+It is:
```

Comment:
```suggestion
It can be:
```

Also, we probably should inform users if some units are not supported by JaCoCo. Or are they all supported now?